plugin/plugin.go

```diff
@@ -21,27 +25,77 @@ func New(policy string) validator.Plugin {
 }
 
 func (p *plugin) Validate(ctx context.Context, req *validator.Request) error {
-
-	var droneConfig DroneConfig
-
-	err := yaml.Unmarshal([]byte(req.Config.Data), &droneConfig)
+	f := bytes.NewBufferString(req.Config.Data)
+	resources, err := Parse(f)
 	if err != nil {
 		return err
 	}
+	for _, resource := range resources {
 
-	r := rego.New(
-		rego.Query("deny = data.drone.validation.deny; msg = data.drone.validation.out"),
-		rego.Load([]string{p.policyPath}, nil),
-		rego.Input(droneConfig))
+		r := rego.New(
+			rego.Query("deny = data.drone.validation.deny; msg = data.drone.validation.out"),
+			rego.Load([]string{p.policyPath}, nil),
+			rego.Input(resource))
 
-	rs, err := r.Eval(ctx)
-	if err != nil {
-		return err
+		rs, err := r.Eval(ctx)
+		if err != nil {
+			return err
+		}
+		if rs[0].Bindings["deny"] == true {
+			message := fmt.Sprintf("pipeline %v", rs[0].Bindings["msg"])
+			return errors.New(message)
+		}
+
+	}
+	return nil
+}
+
+func Parse(r io.Reader) ([]*DroneConfig, error) {
```

Comment:
I think you can probably simplify this with either a string splitting approach or using a yaml decoder to process an `io.Reader`.

See [how helm does this](https://github.com/helm/helm/blob/v3.5.4/pkg/releaseutil/manifest.go#L36-L60) with string splitting, or [here is an example of using a yaml decoder](https://github.com/go-yaml/yaml/blob/v3/decode_test.go#L891-L907).